### PR TITLE
Implement `pygame.image.load_sized_svg`

### LINKED ISSUE #2600: API proposal for scaled (svg) image loading
**Description**

I propose the addition of a new kwarg `scale` (default: `None`) to `pygame.image.load` that takes a size, and makes sure the output surface is in that size (if it is not None).

The idea is that if the source file is something like an SVG, we use the new SDL_image 2.6.0+ functionality that rasterizes the SVG at the given size, so the output looks clean even at arbitrarily large resolutions. Additionally, if the source is not an SVG, I'd like something like `pygame.transform.smoothscale` applied behind the scenes in the image loading function, so this scale argument is generic and not constrained by the format of the source file passed.

I'm gonna work on this proposal if it is accepted, I'm also open to discussing alternatives in this issue

### PR DESCRIPTION
**Pull request by ankith26**
closes #2600

### CODE REVIEW

File: docs/reST/ref/image.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ formats.
 
    * ``QOI``
 
-   * ``SVG`` (limited support, using Nano SVG)
+   * ``SVG``
```
Comment (Starbuck5) [inline docs/reST/ref/image.rst:40]: It’s limited support because it can’t rasterize text, for example.

Comment (ankith26) [reply docs/reST/ref/image.rst:40]: Oh, I see, will re-add that info then

File: docs/reST/ref/image.rst [hunk 2]
```diff
@@ -109,6 +109,28 @@ following formats.
...
+   argument. The ``file`` argument can be either a filename, a Python file-like
+   object, or a pathlib.Path.
+
+   The usage of this function for handling SVGs is recommended, as calling the
+   regular ``load`` function and then scaling the returned surface would not
+   preserve the quality that an SVG can provide.
+
+   It is to be noted that this function does not return a surface whose
+   dimensions exactly match the ``size`` argument. This function preserves
+   aspect ratio, so the returned surface could be smaller along atmost one
```
Comment (oddbookworm) [inline docs/reST/ref/image.rst:127]: Typo? `atmost` -> `at most`

**Review by oddbookworm — APPROVED** [review]
Just one small nitpick that won’t prevent me from approving

**Review by Starbuck5 — APPROVED** [review]
I haven't tested this out locally, but everything seems straightforward enough.

Should the docs say "if not on SDL_image 2.6.0 or above, a pygame.error will be raised" ?

**Pull request merged**